pkg/shortnames/shortnames.go

```diff
@@ -246,6 +250,21 @@ func Resolve(ctx *types.SystemContext, name string) (*Resolved, error) {
 	}
 	resolved.systemContext = ctx
 
+	// Resolve to docker.io only if enforced by the caller (e.g., Podman's
+	// Docker-compatible REST API).
+	if ctx.ResolveShortNamesToDockerHub {
+		named, err := reference.ParseNormalizedNamed(name)
```

Comment:
I think this should only happen after the `!isShort` case, to make sure truly fully-qualified names behave exactly the same everywhere. We can re-normalize at that point (similarly to what `parseUnnormalizedShortName`) does but we shouldn’t re-parse the string input from scratch.